### PR TITLE
Tests for README examples

### PR DESCRIPTION
**Pull request by lmmx**
This PR adds a folder of examples (taken from the README) as standalone Python modules and then adds them to the test suite using `pytest.importorskip` (broken out of previous PR #9 as requested)

- Note: the first commit of this PR is just to pick up where #17 left off

### CODE REVIEW

File: examples/rich_versions.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+
+def uses_rich(a, b):
+    from rich import print
+    from importlib import metadata
+
+    version = metadata.version("rich")
+
+    print(f"hello from rich=={version}")
+    return a + b
```
Comment (koaning) [inline examples/rich_versions.py:14]: This is a nice touch to make the example more clear! Feel free to update the example on the README.md with this one.

**Review by koaning — COMMENTED** [review]
I am not 100% sure about this one. Feels like the gist of the idea is pretty clear on the README.md and I worry that having an examples folder that is outside of the tests folder (but still part of the test suite) adds a bit of bloat that might make it hard to remember what is going on if I do not touch the repository for a year. 

I might be missing a few benefits you had in mind tho, could you elaborate on the PR on this?

**Comment by lmmx** [comment]
That's fair enough, I personally like the pattern of an `examples` dir with self-contained standalone examples I can just run, the same as a `demo.py`. You obviously wouldn't have many `demo.py` files in the root dir so naturally they end up in a folder, and `examples` just feels like a natural name for it to me.

I guess since we already have the concept of "assets" as standalone non-test files (but rather Python programs used by tests) it's fine for them to go in there instead.

File: examples/rich_versions.py (hunk 1 above)
Comment (lmmx) [reply examples/rich_versions.py]: Contributed in 46e130d !

**Comment by lmmx** [comment]
I gathered all of these into a single test file (`examples_test.py` for want of a better name), so now this PR only adds one new test module and one new test 'asset' (`some_script.py` in-keeping with the README's example)

**Review by koaning — APPROVED** [review]

**Comment by koaning** [comment]
That should be all of them. Thanks!

**Comment by lmmx** [comment]
Thanks for chaperoning them in! 😃

**Pull request merged**